packages/react-theme-provider/src/makeStyles.ts

```diff
@@ -35,32 +39,33 @@ const graphSet = (graphNode: Map<any, any>, path: any[], value: any) => {
   graphNode.set(path[path.length - 1], value);
 };
 
-let _seed = 0;
-
-// If the stylesheet reset call is made, invalidate the cache keys.
-Stylesheet.getInstance().onReset(() => _seed++);
-
 /**
  * Registers a css object, optionally as a function of the theme.
  *
  * @param styleOrFunction - Either a css javascript object, or a function which takes in `ITheme`
  * and returns a css javascript object.
  */
 export function makeStyles<TStyleSet extends { [key: string]: IStyle }>(
-  styleOrFunction: TStyleSet | ((theme: ITheme) => TStyleSet),
-): () => { [key in keyof TStyleSet]: string } {
+  styleOrFunction: TStyleSet | ((theme: Theme) => TStyleSet),
+): (theme?: Theme, renderer?: StyleRenderer) => { [key in keyof TStyleSet]: string } {
   // Create graph of inputs to map to output.
   const graph = new Map();
 
-  return () => {
+  return (theme?: Theme, renderer?: StyleRenderer) => {
     const win = useWindow();
-    const theme = useCustomizationSettings(['theme']).theme;
-    const path = theme ? [_seed, win, theme] : [_seed, win];
+
+    // Expected: theme is either always provided or never.
+    theme = theme || useTheme();
+    renderer = (renderer || useStyleRenderer()) as StyleRenderer;
```

Comment:
```suggestion
    renderer = (renderer || useStyleRenderer());
```

is `as StyleRenderer` needed here?

Reply:
only in TS 3.7. In 4.0.2, no. :-/

Reply:
quite confusing.